src/components/link/link.tsx

```diff
@@ -138,6 +153,15 @@ export class Link implements InteractiveComponent {
   /** the rendered child element */
   private childEl: HTMLAnchorElement | HTMLSpanElement;
 
+  /**
+   * handle clicks on child element
+   *
+   * @param event
+   */
+  private childElClickHandler = (event: PointerEvent): void => {
+    event.stopPropagation();
```

Comment:
You still need to determine if the click came from the forwarded click or not. Only stop propagation if it's internal.